code/workspaces/web-app/src/components/common/PromisedContentWrapper.js

```diff
@@ -1,22 +1,99 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { makeStyles } from '@material-ui/core';
 import CircularProgress from '@material-ui/core/CircularProgress';
 
-const PromisedContentWrapper = ({ children, promise, fetchingClassName, completeClassName }) => (
-  promise.fetching
-    ? <div className={fetchingClassName}><CircularProgress /></div>
-    : <div className={completeClassName}>{children}</div>
-);
+const useStyles = makeStyles(() => ({
+  fullSizeBase: {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+  fullWidth: {
+    width: '100%',
+  },
+  fullHeight: {
+    height: '100%',
+  },
+}));
+
+const PromisedContentWrapper = ({ children, promise, className, fetchingClassName, fullWidth = true, fullHeight = false, loadingSize = 40 }) => {
```

Comment:
I'm inclined to agree with the code warning in this case. Just extracting some functions should tidy this up.